lib/src/utils.dart

```diff
@@ -241,10 +241,6 @@ Future pumpEventQueue([int times = 20]) {
   return new Future.delayed(Duration.ZERO, () => pumpEventQueue(times - 1));
 }
 
-/// Like [new Future], but avoids dartbug.com/11911 by using async/await under
-/// the covers.
-Future<T> newFuture<T>(T callback()) async => await callback();
```

Comment:
why not change this to:

```Future<T> newFuture<T>(T callback()) => new Future.microtask(callback);```

Reply:
I don't have a strong opinion. I think it masks intent but it would make the diff smaller.

Reply:
Looking at the original bug it looks like this was done intentionally to ensure consistency since  the different `Future` constructors are inconsistent about how they schedule tasks.

Reply:
OK, keeping the `newFuture` utility now and updated the doc comment. PTAL